### PR TITLE
chore: Do not print sensitive headers

### LINKED ISSUE #5684: Elasticsearch sink logs credentials in debug mode
### Vector Version

```
0.11.1
```

### Vector Configuration File

```toml
# Set global options
    data_dir = "/var/lib/vector"

    # inputs
    [sources.input]
    type = "kafka"

    [sources.metrics]
    type = "internal_metrics"


    # in place mutations
    [transforms.parse_input]
    type = "json_parser"
    inputs = ["input"]
    drop_invalid = true


    # outputs
    [sinks.elasticsearch]
    type = "elasticsearch"
    inputs = ["parse_input"]
    index = "logstash-input-%Y.%m.%d"
    host = "https://elasticsearch-master:9200"
    [sinks.elasticsearch.batch]
    max_size = 5490000
    timeout_secs = 1
    [sinks.elasticsearch.auth]
    strategy = "basic"
    password = "${ELASTIC_PASSWORD}"
    user = "${ELASTIC_USERNAME}"

    [sinks.metrics_out]
    type = "prometheus_exporter"
    inputs = ["metrics"]
    address = "0.0.0.0:8080"
    default_namespace = "detections"
```

### Expected Behavior

I believe that the elasticsearch sink should have sanitized the header "authorization": "Basic <my_credentials_base_64_encoded>" in debug mode.

### Actual Behavior

It logged the entire header which included my credentials for my elasticsearch cluster

### Example Data

```
DEBUG http: vector::internal_events::http_client: Sending HTTP request. uri=https://elasticsearch-master:9200/_cluster/health method=GET version=HTTP/1.1 headers={"authorization": "Basic <my_credentials>", "user-agent": "Vector/0.11.1 (v0.11.1 x86_64-unknown-linux-musl 2020-12-17)", "accept-encoding": "identity"} body=[empty]
```

### PR DESCRIPTION
**Pull request by juchiast**
close #5684 

I've scanned the list of standard headers in https://docs.rs/http/0.2.2/http/header/index.html and picked some potentially sensitive headers.

### CODE REVIEW

File: src/internal_events/http_client.rs [hunk 1]
```diff
@@ -9,14 +12,30 @@ pub struct AboutToSendHTTPRequest<'a, T> {
     pub request: &'a Request<T>,
 }
 
+fn remove_sensitive(headers: &HeaderMap<HeaderValue>) -> HeaderMap<HeaderValue> {
+    let mut headers = headers.clone();
+    let redacted = HeaderValue::from_static("<redacted>");
```
Comment (fanatid) [inline src/internal_events/http_client.rs:17]: We can use [HeaderValue::set_sensitive](https://docs.rs/http/0.2.1/http/header/struct.HeaderValue.html#method.set_sensitive)

Comment (MOZGIII) [reply src/internal_events/http_client.rs:17]: I agree, let's use `HeaderValue::set_sensitive` here!

Comment (juchiast) [reply src/internal_events/http_client.rs]: Thanks!

**Review by fanatid — APPROVED** [review]
:+1: 
It would be nice to avoid extra clone but not sure that there is a simple way to do so.

**Comment by juchiast** [comment]
@fanatid I checked and the function will not be called if it is not `debug` level, so it should not affect performance.

**Review by MOZGIII — APPROVED** [review]

**Pull request merged**